CMakeLists.txt

```diff
@@ -102,15 +115,38 @@ set(TARGET_NAME slim)
 file(GLOB_RECURSE SLIM_SOURCES ${PROJECT_SOURCE_DIR}/core/*.cpp ${PROJECT_SOURCE_DIR}/eidos/*.cpp)
 add_executable(${TARGET_NAME} ${SLIM_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/core" "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
-target_link_libraries(${TARGET_NAME} PUBLIC)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
 
 set(TARGET_NAME eidos)
 file(GLOB_RECURSE EIDOS_SOURCES  ${PROJECT_SOURCE_DIR}/eidos/*.cpp  ${PROJECT_SOURCE_DIR}/eidostool/*.cpp)
 add_executable(${TARGET_NAME} ${EIDOS_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
+
+set(TARGET_NAME eidoslib)
```

Comment:
This might be a bit scrambled...?  Note that there's a target "eidos" just above.  The "eidoslib" target should probably build a library using /eidos/*.cpp, without /eidostool/*.cpp.  The "eidos" target should probably then depend upon the "eidoslib" target with the addition of /eidostool/*.cpp; the file in /eidostool/main.cpp just supplies a main() function for the command-line eidos tool.  That main() function should not be part of the build of the eidos library, nor should it be a part of the build of QtSLiM as it seems to be now (?).

Reply:
This is not really needed (as mentioned below) if a full rebuild is needed, so this target can just go away).